### PR TITLE
chore(forbidden-identifiers): detect incorrect line endings

### PR DESCRIPTION
**Pull request by DevVersion**
* Removes stale scope checking (from `@angular2-material/$COMPONENT` times)
* Adds support for error grouping (useful when having a file full of invalid line endings)
* Moves chain functions into external functions to easily 

understand procession.

> Here you can see the error grouping - Notice that we don't show any special message for the line endings (because to keep the script small and easy to maintain)

### CODE REVIEW

File: scripts/ci/forbidden-identifiers.js [hunk 1]
```diff
@@ -47,69 +46,105 @@ findTestFiles()
...
+
+    return true;
   });
 
+  /** Checks whether the given error is similar to the previous one. */
+  function isSimilarError(error) {
+    return initialError.fileName === error.fileName &&
+           initialError.statement === error.statement &&
+           previousLine === (error.lineNumber - 1);
+  }
```
Comment (DevVersion) [inline scripts/ci/forbidden-identifiers.js]: I know this is not very pretty, but we don't have any Lodash helpers here, and this seems to be an elegant approach.

**Comment by jelbourn** [comment]
@hansl can you take a look?

**Comment by DevVersion** [comment]
@hansl Do you have time to take a quick look at this? :)

**Comment by hansl** [comment]
Hi @DevVersion,

Sorry, this fell off a bit. Looking at it now.

**Comment by hansl** [comment]
LGTM for now.

@DevVersion What I'd like to see though is moving away from a `forbidden-identifier` file and using custom `tslint` rules instead. It would be better and less hacky (since reporting is done by tslint, for example). See documentation here: https://www.npmjs.com/package/tslint#custom-rules

What do you think?

**Comment by DevVersion** [comment]
@hansl - Great idea. I can do a follow-up PR to this one with the custom TS-Lint rules-

**Comment by hansl** [comment]
Yes, a follow up PR. This one is okay (for now).

**Pull request merged**